tools/uart-flash-server/ufserver.c

```diff
@@ -0,0 +1,418 @@
+/* ufserver.c
+ *
+ * UART flash server
+ *
+ * Run on HOST machine to export an emulated external, non-volatile memory.
+ *
+ * Copyright (C) 2020 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include "wolfboot/wolfboot.h"
+#include "hal.h"
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <termios.h>
+#include <errno.h>
+#include <string.h>
+#include <libgen.h>
+
+
+#define CMD_HDR_WOLF  'W'
+#define CMD_HDR_VER   'V'
+#define CMD_HDR_WRITE 0x01
+#define CMD_HDR_READ  0x02
+#define CMD_HDR_ERASE 0x03
+#define CMD_ACK       0x06
+
+#define FIRMWARE_PARTITION_SIZE 0x4000
+#define SWAP_SIZE 0x4000
+#define UART_BITRATE 460800
+
+const char msgSha[]         = "Verifying SHA digest...";
+const char msgReadUpdate[]  = "Fetching update blocks ";
+const char msgReadSwap[]    = "Reading SWAP blocks    ";
+const char msgWriteUpdate[] = "Writing backup blocks  ";
+const char msgWriteSwap[]   = "Writing SWAP blocks    ";
+const char msgEraseUpdate[] = "Erase update blocks    ";
+const char msgEraseSwap[]   = "Erase swap blocks      ";
+
+extern uint16_t wolfBoot_find_header(uint8_t *haystack, uint16_t type, uint8_t **ptr);
+
+const char blinker[]="-\\|/";
+
+void printmsg(const char *msg)
+{
+    static const char *cur_msg = NULL;
+    static int b_idx = 0;
+    if (cur_msg != msg) {
+        cur_msg = msg;
+    }
+    b_idx++;
+    if (b_idx > 3)
+        b_idx = 0;
+    printf("\r[%c] %s\t\t\t", blinker[b_idx], msg);
+    fflush(stdout);
+}
+
+
+int hal_flash_write(uint32_t address, const uint8_t *data, int len)
+{
+    (void)address;
+    (void)data;
+    (void)len;
+    return -1;
+}
+
+int hal_flash_erase(uint32_t address, int len)
+{
+    (void)address;
+    (void)len;
+    return -1;
+}
+void hal_flash_unlock(void)
+{
+}
+void hal_flash_lock(void)
+{
+}
+
+#include <termio.h>
```

Comment:
I sent a patch via email to fix building this on the Mac. 
Don't include termio.h and linux/serial.h and change the B() macro to just return the value. And exclude the "custom divisor" chunk for Mac, since it doesn't have `struct serial_struct`.

Reply:
patch applied thanks

Reply:
I re-added the includes under `#ifndef __MACH___` so that it builds on GNU again